src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -216,10 +217,32 @@ public override Task SetParametersAsync(ParameterView parameters)
                     $"{nameof(Forms.EditContext)} dynamically.");
             }
 
+            SetAdditionalAttributesIfValidationFailed();
+
             // For derived components, retain the usual lifecycle with OnInit/OnParametersSet/etc.
             return base.SetParametersAsync(ParameterView.Empty);
         }
 
+        private void OnValidateStateChanged(object sender, ValidationStateChangedEventArgs eventArgs)
+        {
+            SetAdditionalAttributesIfValidationFailed();
+
+            StateHasChanged();
+        }
+
+        private void SetAdditionalAttributesIfValidationFailed()
+        {
+            if (!EditContext.GetValidationMessages(FieldIdentifier).Any())
```

Comment:
Would it help if we added a `IsValid` \  `HasValidationMessages` to `EditContext`?

Reply:
I think it makes sense to add `IsValid` for sure, and not in this particular case only. I've seen customers in other comments also using this technique to check the validation state.